### ISSUE TITLE
Feature request: Pipeline Triggers on Git branch deletion and Git pull request state change

### ISSUE THREAD

**Issue by jfillo** [spinnaker/spinnaker#3080]
### Feature Request:
Allow Pipelines to be triggered on Git Branch deletion and Git Pull Request state changes

### Cloud Provider(s):
Any

### Feature Area: 
Pipeline Triggers 

### Description:
It would be useful to have the ability to trigger pipelines when a git branch is delete or a pull request is moved to an open or closed state.

These triggers would allow users to create pipelines that will automatically tear down application environments that are associated with that branch that no longer exists or pull request that has been closed.

### Additional Info:
I am currently doing this by running my own webserver that responds to git hooks and translates the git hook payload into a spinnaker webhook pipeline trigger. I'd love to be able to drop this and instead have Git and Spinnaker communicate to each other directly.

**Comment by sbwsg** [comment]
Thanks for the feature request! This appears to be a duplicate of https://github.com/spinnaker/spinnaker/issues/1873 so I'm going to close it. It's worth keeping these coalesced in single issues to allow contributors to easily read and review the ongoing conversation.

If you feel strongly that this feature request is different enough to warrant its own issue then please re-open with a description of how it differs.

**Comment by adrian-gierakowski** [comment]
@sbwsg I don't think this is a duplicate of #1873. That issue is about adding support for generic git repos. This on the other hand is about being able to trigger pipelines in response to branch/PR related events (presumably when using one of the currently supported git providers).

**Comment by jfillo** [comment]
@adrian-gierakowski is correct. This issue fell off my radar but it is a completely separate feature request. @sbwsg should we resubmit the request or can this be re-opened?

**Comment by spinnakerbot** [comment]
This issue hasn't been updated in 45 days, so we are tagging it as 'stale'. If you want to remove this label, comment:

> @spinnakerbot remove-label stale

**Comment by spinnakerbot** [comment]
This issue is tagged as 'stale' and hasn't been updated  in 45 days, so we are tagging it as 'to-be-closed'. It will be closed in 45 days unless updates are made. If you want to remove this label, comment:

> @spinnakerbot remove-label to-be-closed

**Comment by spinnakerbot** [comment]
This issue is tagged as 'to-be-closed' and hasn't been updated  in 45 days, so we are closing it. You can always reopen this issue if needed.

**Comment by adrian-gierakowski** [comment]
Could this be reopened?

**Comment by mhuisman** [comment]
Bump for fuller-featured git webhook integration.  Is this waiting on someone enthusiastic enough to volunteer?

**Comment by ghostsquad** [comment]
Please reopen

**Comment by k1nger** [comment]
This needs to be re-opened. We should be able to access git webhook parameters:

https://developer.github.com/enterprise/2.20/webhooks/
